docs/whitepaper.tex

```diff
@@ -247,38 +247,98 @@ \section{Tests}
 
 \todo{Should we write Apropos-Tx tests?}
 
-\subsection{Unit tests}
+\subsection{Components}
+
+Tests are broken down by the onchain component they test.
+
+\subsubsection{Config utxo tests}
+
+NFT tests
 
-The following intended behaviors need to work:
 \begin{itemize}
-  \item Initializing a protocol and pool works
-  \item queries accurately reflect how the pool was setup
-  \item valid swaps work
-  \item the queries correctly report which swaps would be valid
-  \item queries accurately reflect the change from a swap
-  \item valid liquidity changes work
-  \item queries reflect the liquidity changes
-  \item admin can kill a pool
-  \item liquidity can be extracted from a dead pool
+	\item Minting an NFT actually produces an NFT
+	\item Minting an NFT fails if the seed Tx is a reference input
+	\item Trying to mint the NFT twice fails
+	\item The seed Tx has been spent after minting an NFT
+	\item The NFT can't be burned
 \end{itemize}
 
-Additional tests:
+Config utxo address test
+
+\begin{itemize}
+	\item The config utxo can't be spent
+\end{itemize}
+
+\subsubsection{Liquidity Token Minting Policy tests}
+
+\begin{itemize}
+	\item Can mint $\sqrt{A_0B_0}$ liquidity tokens on pool open
+	\item Can't mint a different pool's liquidity token on pool open
+	\item Can't mint multiple liquidity token types on pool open
+	\item Can mint liquidity tokens corresponding to the pool on deposit
+	\item Can burn liquidity on withdrawal
+	\item Can't mint liquidity for a different pool than the pool being spent
+	\item Can't burn liquidity from a different pool than the pool being spent
+	\item Can't mint liquidity for pool at any address other than the pool address
+\end{itemize}
+
+\subsubsection{Pool id Token Minting Policy tests}
+
+\begin{itemize}
+	\item Non admin can initialize a pool
+	\item Can't under pay for liquidity on open
+	\item Can't open pool with 0 liquidity
+	\item Can't open pool with 0 of either exchange token
+		%% This is sorta redundant cause the liquidity needs to be
+		%% wrong or 0
+	\item A utxo without the config NFT can't be used to open a pool
+	\item Can't mint more than 1 token
+	\item Can't send identifier token to non-pool address
+	\item Can't mint without spending seed Tx
+	\item Can't issue more liquidity than datum reflects
+	\item Can't mint with liquidity not equal to the initial invariant
+	\item Can't remove pool id from address during any type of transaction
+	\item Can't kill a pool with a swap
+	\item Can't kill a pool with a liquidity change
+	\item Can't swap on a dead pool
+	\item Can't add liquidity to a dead pool
```

Comment:
Extra test: Withdraw liquidity in one token class fails after killing the pool.
Note: This can both be tested through a unit test (off-chain) and Apropos.